core/src/main/java/org/mobilitydata/gtfsvalidator/notice/Notice.java

```diff
@@ -40,11 +42,17 @@ public SeverityLevel getSeverityLevel() {
   }
 
   /**
-   * Returns a descriptive type-specific name for this notice.
+   * Returns a descriptive type-specific name for this notice based on the class simple name.
    *
    * @return notice code, e.g., "foreign_key_error".
    */
-  public abstract String getCode();
+  public String getCode() {
+    String code = CaseFormat.LOWER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, getClass().getSimpleName());
+    if (code.endsWith(NOTICE_SUFFIX)) {
```

Comment:
Please use org.apache.commons.lang3.StringUtils.removeEnd function.

I am encouraging you to google for ready library functions to perform basic operations in Java. That will make our code more idiomatic and readable.

Reply:
Changes have been provided in https://github.com/MobilityData/gtfs-validator/pull/788/commits/ddc44ccd9ee4ff238ddda1bfe907615bbe27b962